test/unit/org/apache/cassandra/io/sstable/SSTableReaderTest.java

```diff
@@ -324,24 +331,80 @@ public void testGetPositionsKeyCacheStats()
         CompactionManager.instance.performMaximal(store, false);
 
         SSTableReader sstable = store.getLiveSSTables().iterator().next();
+        // existing, non-cached key
         sstable.getPosition(k(2), SSTableReader.Operator.EQ);
+        assertEquals(1, sstable.getKeyCacheRequest());
         assertEquals(0, sstable.getKeyCacheHit());
-        assertEquals(1, sstable.getBloomFilterTruePositiveCount());
```

Comment:
Even if there is a more in depth check of `BloomFilterTruePositiveCount` we still could keep that kind of assertions in this test.

Reply:
I do not like having BF stats asserts in `testGetPositionsKeyCacheStats` because they make the test scope vague (strictly speaking BF stats is irrelevant to this test). Now we have two separate tests with clear scope (KC stats and BF stats) and I see no benefits of having the same asserts in two tests. In my opinion, having asserts covering the same functionality in different tests makes working with issues harder (e.g. multiple tests fail for the same issue, etc).

It is not a problem to recover these asserts, I just want to ensure that's what we really want. @blerer @blambov what is your opinion on my reasoning?

Reply:
Fine with me.